    @Test
    public final void getSource() {
        try {
            root1.getSource(2);
            Assert.fail();
        } catch (Exception e) {

        }
        try {
            root2.getSource(2);
            Assert.fail();
        } catch (Exception e) {

        }
        try {
            root3.getSource(2);
            Assert.fail();
        } catch (Exception e) {

        }

        try {
            Assert.assertNotNull(root3.getSource(0));
            Assert.fail();
        } catch (Exception e) {

        }
        try {
            Assert.assertNotNull(root2.getSource(0));
            Assert.fail();
        } catch (Exception e) {

        }
        try {
            Assert.assertNotNull(root1.getSource(0));
            Assert.fail();
        } catch (Exception e) {

        }
    }

